include/fixed_containers/reflection.hpp

```diff
@@ -19,120 +20,216 @@ namespace fixed_containers::reflection_detail
 template <typename T>
 concept Reflectable = std::is_class_v<T> && ConstexprDefaultConstructible<T>;
 
-class DepthTracker
+enum class LayerType
 {
-    int depth_{-1};
+    ENCLOSING_FIELD,
+    BASE_CLASS,
+};
+
+struct FieldAsString
+{
+    std::string_view type_name;
+    std::string_view name;
+};
+
+inline constexpr std::string_view NULL_FIELD_TYPE_NAME{};
+
+template <std::size_t MAXIMUM_LAYERS = 32>
+class LayerTracker
+{
+    FixedStack<LayerType, MAXIMUM_LAYERS> layer_type_stack_;
+    FixedStack<FieldAsString, MAXIMUM_LAYERS> nesting_stack_;
+    FixedStack<std::string_view, MAXIMUM_LAYERS> inheritance_stack_;
+
+public:
+    constexpr LayerTracker()
+      : layer_type_stack_{}
+      , nesting_stack_{}
+      , inheritance_stack_{}
+    {
+    }
 
 public:
-    [[nodiscard]] constexpr int depth() const
+    [[nodiscard]] constexpr bool is_null_layer() const { return layer_type_stack_.empty(); }
+    [[nodiscard]] constexpr FieldAsString current_enclosing_field() const
     {
-        assert(depth_ != -1);
-        return depth_;
+        return nesting_stack_.top();
+    }
+    [[nodiscard]] constexpr std::string_view current_providing_base_class() const
+    {
+        return inheritance_stack_.empty() ? NULL_FIELD_TYPE_NAME : inheritance_stack_.top();
```

Comment:
Why use a sentinel value instead of just `std::optional`?

Reply:
I went back and forth on this (and also for the same issue for `enclosing_field_name()` which is empty for the most external layer). Might go for optional.

Also, inconsistency: https://github.com/teslamotors/fixed-containers/pull/58/files#diff-f67f3de59b958f663c7f020e0cdf5d23c952ad56b4abe200caec3c3f60d9ec47R72
The other instance of this uses a hard-coded `""`.

Reply:
https://github.com/teslamotors/fixed-containers/pull/61